src/main/java/me/ryanhamshire/griefprevention/task/CleanupUnusedClaimsTask.java

```diff
@@ -83,50 +82,35 @@ public void run() {
 
-                            claim.removeSurfaceFluids(null);
-                            claimManager.deleteClaim(claim);
-
-                            // if configured to do so, restore the land to natural
-                            if (GriefPreventionPlugin.instance.claimModeIsActive(worldProperties, ClaimsMode.Creative) || activeConfig
-                                .getConfig().claim.claimAutoNatureRestore) {
-                                GriefPreventionPlugin.instance.restoreClaim(claim, 0);
-                            }
-
-                            GriefPreventionPlugin.addLogEntry(" " + claim.getOwnerName() + "'s new player claim " + "'" + claim.id + "' expired.",
-                                CustomLogEntryTypes.AdminActivity);
-                        }
+                    int claimExpiration = 0;
+                    EventContextKey context = GPContextKeys.PLAYER_CLAIM_EXPIRED;
+                    if (claim.getArea() <= areaOfDefaultClaim) {
+                        claimExpiration = playerData.optionClaimExpirationChest;
+                        context = GPContextKeys.CHEST_CLAIM_EXPIRED;
+                    } else if (claim.isBasicClaim()) {
+                        claimExpiration = playerData.optionClaimExpirationBasic;
+                    } else if (claim.isSubdivision()) {
+                        claimExpiration = playerData.optionClaimExpirationSubdivision;
+                    } else if (claim.isTown()) {
+                        claimExpiration = playerData.optionClaimExpirationTown;
                     }
 
-                    Double
-                        claimExpirationBasic =
-                        GPOptionHandler.getClaimOptionDouble(playerData.getPlayerSubject(), claim, GPOptions.CLAIM_EXPIRATION_BASIC, playerData);
-                    if (claimExpirationBasic > 0) {
-                        if (claimLastActive.plus(Duration.ofDays(claimExpirationBasic.intValue()))
-                            .isBefore(Instant.now())) {
-                            Sponge.getCauseStackManager().addContext(GPContextKeys.PLAYER_CLAIM_EXPIRED, GriefPreventionPlugin.instance.pluginContainer);
+                    if (claimExpiration > 0 && claimLastActive.plus(Duration.ofDays(claimExpiration)).isBefore(Instant.now())){
+                        Sponge.getCauseStackManager().addContext(context, GriefPreventionPlugin.instance.pluginContainer);
 
-                            claimManager.deleteClaim(claim);
-                            GriefPreventionPlugin.addLogEntry("Removed " + claim.getOwnerName() + "'s unused claim @ "
-                                                              + GriefPreventionPlugin.getfriendlyLocationString(claim.getLesserBoundaryCorner()),
-                                CustomLogEntryTypes.AdminActivity);
+                        claim.removeSurfaceFluids(null);
```

Comment:
Not 100% sure if this was meant to be run every time as previously only chest claim expirations executed this method.